lighthouse-core/audits/script-treemap-data.js

```diff
@@ -185,25 +184,28 @@ class ScriptTreemapDataAudit extends Audit {
     const bundles = await JsBundles.request(artifacts, context);
     const duplicationByPath = await ModuleDuplication.request(artifacts, context);
 
-    for (const scriptElement of artifacts.ScriptElements) {
-      if (!scriptElement.src) continue;
+    for (const script of artifacts.Scripts) {
+      if (script.url === artifacts.URL.finalUrl) continue; // Handled above.
 
-      const name = scriptElement.src;
-      const bundle = bundles.find(bundle => scriptElement.src === bundle.script.src);
-      const scriptCoverages = artifacts.JsUsage[scriptElement.src] || [];
-      if (!bundle && scriptCoverages.length === 0) {
+      const name = script.name;
+      const bundle = bundles.find(bundle => script.scriptId === bundle.script.scriptId);
+      const scriptCoverage = /** @type {Omit<LH.Crdp.Profiler.ScriptCoverage, "url"> | undefined} */
+        (artifacts.JsUsage[script.scriptId]);
+      if (!bundle && !scriptCoverage) {
         // No bundle and no coverage information, so simply make a single node
         // detailing how big the script is.
 
         nodes.push({
           name,
-          resourceBytes: scriptElement.src.length,
+          resourceBytes: script.length || 0,
         });
         continue;
       }
 
-      const unusedJavascriptSummary = await UnusedJavaScriptSummary.request(
```

Comment:
previously this just worked because we'd fall back to an empty array above. now, it makes more since to just do this  conditionally